### ISSUE TITLE
Training with multiple caption options

### ISSUE THREAD

**Issue by schrum2** [schrum2/MarioDiffusion#177]
Regarding the idea in #166 about training with multiple captions.

Our code already supports a variant of this (there is a data augmentation option already). The way it currently works is that the structured captions are broken down into their phrases and re-ordered, so each scene gets trained on different orderings of the same phrases.

This new option is more drastic, but needs to be related. First, there is the question of how to store the data. The current convention for the json training files is to store an attribute named "caption" with the caption. Let's keep this in place. However, if we want a dataset with multiple captions, then the additional caption attributes will be "caption1", "caption2", etc. (but plain "caption" is still present).

Then, in train_diffusion (and maybe in certain parts of the level dataset) you have support for augmentation via caption selection (as opposed to phrase shuffling, which is already supported). I think that "augmentation" in the current code might also correspond to mirrored scenes. Basically, if we are selecting from multiple captions, then that is the only augmentation we want ... don't combine it with other types of augmentation.

If the code/dataset is aware of this setting, then it can load all captions, and select one of the available ones at random each time data is selected for training.

The nice thing about this is that we can test it in the following way: take our existing captions, and make a dataset that contains different random phrase orderings. In other words, training on this new dataset should be equivalent to what we are already doing (augmenting by shuffling), so it should work fine and be a good test case.

**Comment by fwilson12** [comment]
Just began a test run training a model with multi-captioned smb1 data; I'll paste the commands I used to generate the dataset and train the model below for future reference in case we want to make a batch file for this:

```
python create_multiple_caption_dataset.py --input datasets\SMB1_LevelsAndCaptions-regular-train.json --output datasets\SMB1_LevelsAndCaptions-regular-multi-train.json --num_captions 5

python create_multiple_caption_dataset.py --input datasets\SMB1_LevelsAndCaptions-regular-test.json --output datasets\SMB1_LevelsAndCaptions-regular-multi-test.json --num_captions 5

python create_multiple_caption_dataset.py --input datasets\SMB1_LevelsAndCaptions-regular-validate.json --output datasets\SMB1_LevelsAndCaptions-regular-multi-validate.json --num_captions 5

python train_diffusion.py --multiple_captions --augment --mlm_model_dir SMB1-MLM-regular0 --pkl datasets\SMB1_Tokenizer-regular.pkl --json datasets\SMB1_LevelsAndCaptions-regular-multi-train.json --val_json datasets\SMB1_LevelsAndCaptions-regular-multi-validate.json --text_conditional --num_epochs 500 --output_dir SMB1-conditional-regular0-multicap
```

**Comment by fwilson12** [comment]
results look good even after only 140 epochs;  

<img width="504" height="342" alt="Image" src="https://github.com/user-attachments/assets/c3dcdf20-9634-4db8-be6d-47143ed3e7df" />
